service/s3/s3crypto/aes_helper.go

```diff
@@ -0,0 +1,46 @@
+package s3crypto
+
+import "bytes"
+
+func padAESKey(key []byte) []byte {
```

Comment:
should document where the padding is defined for AES. Will make understanding the logic easier later.